subsys/usb/usb_device.c

```diff
@@ -1197,26 +1198,28 @@ static int class_handler(struct usb_setup_packet *pSetup,
 {
 	size_t size = (__usb_data_end - __usb_data_start);
 	const struct usb_if_descriptor *if_descr;
-	struct usb_interface_cfg_data *iface;
+	const struct usb_interface_cfg_data *handlers;
+	const struct usb_cfg_data *cfg = NULL;
+	uint8_t interface_number = sys_le16_to_cpu(pSetup->wIndex);
```

Comment:
This implicitly casts `uint16_t` to `uint8_t`. I think it would be good to add explicit casting here, to avoid warnings/errors in some future. Same story in [line 1236](https://github.com/zephyrproject-rtos/zephyr/pull/26075/files#diff-609952c961f3c77cb7811bedee4bfd2cR1236).